src/main/scala/org/broadinstitute/dsde/firecloud/webservice/UserApiService.scala

```diff
@@ -83,43 +83,43 @@ trait UserApiService extends HttpService with PerRequestCreator with FireCloudRe
         authorizationHeader match {
           // no Authorization header; the user must be unauthorized
           case None =>
-            respondWithErrorReport(Unauthorized, Unauthorized.defaultMessage, requestContext)
-          // browser sent Authorization header; try to query rawls for user status
+            respondWithErrorReport(Unauthorized, "No authorization header in request.", requestContext)
+          // browser sent Authorization header; try to query Sam for user status
           case Some(c) =>
             val pipeline = authHeaders(requestContext) ~> sendReceive
-            val extReq = Get(UserApiService.samRegisterUserURL)
-            pipeline(extReq) onComplete {
+            val samRequest = Get(UserApiService.samRegisterUserURL)
+            pipeline(samRequest) onComplete {
               case Success(response) =>
                 response.status match {
-                  // rawls rejected our request. User is either invalid or their token timed out; this is truly unauthorized
-                  case Unauthorized => respondWithErrorReport(Unauthorized, Unauthorized.defaultMessage, requestContext)
-                  // rawls 404 means the user is not registered with FireCloud
-                  case NotFound => respondWithErrorReport(NotFound, "FireCloud user registration not found", requestContext)
-                  // rawls error? boo. All we can do is respond with an error.
-                  case InternalServerError => respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
-                  // rawls found the user; we'll try to parse the response and inspect it
+                  // Sam rejected our request. User is either invalid or their token timed out; this is truly unauthorized
+                  case Unauthorized => respondWithErrorReport(Unauthorized, "Request rejected by identity service - invalid user or expired token.", requestContext)
+                  // Sam 404 means the user is not registered with FireCloud
+                  case NotFound => respondWithErrorReport(NotFound, "FireCloud user registration not found.", requestContext)
+                  // Sam error? boo. All we can do is respond with an error.
+                  case InternalServerError => respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again. (Error 10)", requestContext)
```

Comment:
for this unexpected error, suggest you pass the root cause back up to the UI? Something like:
```
case ise:InternalServerError => respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again. (Error 10)", ise, requestContext)
```
of course, that signature for `respondWithErrorReport` - a signature that accepts a `Throwable` and uses the throwable when creating its `ErrorReport` - doesn't exist, so you'll have to create it.

Reply:
`InternalServerError` is a `val` on `spray.http.StatusCodes`, not the error itself

I see what you're getting at though, I think it's possible